src/run/run.c

```diff
@@ -336,55 +344,63 @@ static int parse_argv(int argc, char *argv[]) {
                         if (r < 0)
                                 return r;
 
-                        with_timer = true;
+                        arg_with_timer = true;
                         break;
 
                 case ARG_ON_BOOT:
                         r = add_timer_property("OnBootSec", optarg);
                         if (r < 0)
                                 return r;
 
-                        with_timer = true;
+                        arg_with_timer = true;
                         break;
 
                 case ARG_ON_STARTUP:
                         r = add_timer_property("OnStartupSec", optarg);
                         if (r < 0)
                                 return r;
 
-                        with_timer = true;
+                        arg_with_timer = true;
                         break;
 
                 case ARG_ON_UNIT_ACTIVE:
                         r = add_timer_property("OnUnitActiveSec", optarg);
                         if (r < 0)
                                 return r;
 
-                        with_timer = true;
+                        arg_with_timer = true;
                         break;
 
                 case ARG_ON_UNIT_INACTIVE:
                         r = add_timer_property("OnUnitInactiveSec", optarg);
                         if (r < 0)
                                 return r;
 
-                        with_timer = true;
+                        arg_with_timer = true;
                         break;
 
                 case ARG_ON_CALENDAR:
                         r = add_timer_property("OnCalendar", optarg);
                         if (r < 0)
                                 return r;
 
-                        with_timer = true;
+                        arg_with_timer = true;
+                        break;
+
+                case ARG_ON_TIMEZONE_CHANGE:
+                        arg_with_timer = arg_on_timezone_change = true;
```

Comment:
This is not a strong opinion, but I prefer to use `add_timer_property()` or completely drop the function.

If I remember correctly, I added `add_timer_property()` to make the order of the settings in the created transient unit is equivalent to the order of the options specified in the command line.

However, now these new options make the relevant settings in the last...

Reply:
fixed. ptal!